### PR TITLE
#72 Persisting fixture factories on the command line

### PR DESCRIPTION
**Pull request by pabloelcolombiano**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +90,18 @@ Persisting five articles having each three different authors, each with differen
 $article = ArticleFactory::make(5)->with('Authors[3].Address.City.Country', ['name' => 'Kenya'])->persist();
 ```
 
+#### On the command line:
+Factories can also conveniently populate your database in order to test your application on the browser.
```
Comment (pakacuda) [inline README.md:94]: Test your application on the browser ? Shouldn't it be in the command line ?
If not, what is the use case for making this available.
Also the title of the shouldn't the pull request title be 'fixture factories on the command line'. I'm not sure I get what is meant by 'inline' here.
Can you just explain a little more what a use case would be for this ?

Is it : 
* I'm testing a feature in my app with my browser but I would like to have more records to test this 
* So I go to the command line and I create a bunch of records quickly so I can test some feature more easily ?

Once this is clarified, I could go further with the review

Comment (pabloelcolombiano) [reply README.md:94]: There are two main use cases.

The first one is how you described, inserting data in the DB in order to manually test the app. Example: the tester wants to test the pagination of the articles index page, the provided command will enable to persist 100 of articles, facilitating the testing. Another example: a customer complains that a given query is too slow for a particularly large amount fo data. The developer can easily insert a large set of data aiming at reproducing the artifact.

The second use case enables the developer, with the `--dry-run` option, to easily display the output of her factories.

File: README.md [hunk 2]
```diff
@@ -90,6 +90,18 @@ Persisting five articles having each three different authors, each with differen
...
+The following command will persist 5 articles, each with 3 irish authors, considering that the `ArticleFactory` class features
+a `withThreeIrishAuthors()` method:
+```$xslt
+bin/cake fixture_factories_persist Authors -n 5 -m withThreeIrishAuthors
+```
+The option `--dry-run` or `-d` will display the output without persisting.
+The option `-c` will persist in the connection provided (default is `test`).
+
+The `fixture_factories_persist` command is featured on CakePHP 4 only (open to contribution for CakePHP 3).
+
```
Comment (pakacuda) [inline README.md:104]: We could also add another parameter to allow the use of a string in the with method. This should be useful and very simple to implement

```$xslt
bin/cake fixture_factories_persist Authors -n 5 -m withThreeIrishAuthors -w "Address.City.Country"
```

Which would result in calling : 

```
AuthorsFActory::make(5)->with('Address.City.Country')->withThreeIrishAuthors();
```

File: README.md [hunk 3]
```diff
@@ -90,6 +90,18 @@ Persisting five articles having each three different authors, each with differen
...
+Factories can also conveniently populate your database in order to test your application on the browser.
+The following command will persist 5 articles, each with 3 irish authors, considering that the `ArticleFactory` class features
+a `withThreeIrishAuthors()` method:
+```$xslt
+bin/cake fixture_factories_persist Authors -n 5 -m withThreeIrishAuthors
+```
+The option `--dry-run` or `-d` will display the output without persisting.
+The option `-c` will persist in the connection provided (default is `test`).
+
+The `fixture_factories_persist` command is featured on CakePHP 4 only (open to contribution for CakePHP 3).
```
Comment (pakacuda) [inline README.md:103]: Fine for me if cake 4 only

File: README.md (hunk 2 above)
Comment (pabloelcolombiano) [reply README.md:104]: I agree, this is a good additional feature. Will implement that.

**Pull request merged**